src/systems/detachable_joint/DetachableJoint.cc

```diff
@@ -234,8 +234,75 @@ void DetachableJoint::PreUpdate(
     }
     else
     {
-      modelEntity = _ecm.EntityByComponents(
-          components::Model(), components::Name(this->childModelName));
+      auto entitiesMatchingName = entitiesFromScopedName(
```

Comment:
EntitiesFromScopedName seems like a very expensive function call, with complexity O(P*N*R) where P is the length of the scoped path, N is the number of named entities in the ECM and R is the size of the function output. I'd recommend that we do not use it in PreUpdate/PostUpdate as it could affect performance for large scenes.

I'd prefer if we moved the entire logic for selecting the child model and child model link to the Configure call since if the plugin sdf contains a reference to a model, presumably it should already be present in the world at the time when Configure is called. 

However, that may break existing apps which depend on run-time addition and discovery of the child model, which the existing implementation allows, if only in a restrictive way.

One alternative is to move this logic to a helper function `GetChildModelAndLinkEntities(...)` and call it in Configure first. Then in PreUpdate, we can call `GetChildModelAndLinkEntities(...)` conditionally if `childLinkEntity == kNullEntity || !_ecm.HasEntity(childLinkEntity)` to update the cached entities.

Reply:
Thats a good point. We only run this when we call attach, but reading the implementation of `entitiesFromScopedName` I can see how this can be a problem. Might be worth exploring ways to optimize this in future releases, 

Fixed in [1a7083d](https://github.com/gazebosim/gz-sim/pull/1097/commits/1a7083dfeb05bd50e6358bbbd3377f6a13e0bb4e)